### DISCUSSION TITLE
Allowing rules to call "remotes" API?

### DISCUSSION THREAD

**Discussion by ArcticSnowman** [target/flottbot#522] [Ideas]
Item brought up in #520 was whether you could interrogate the remote to see if the reaction had been responded to previously.  This discussion is to consider approaches to allow check access to the remote APIs.

Two options that come to mind is 

1. The rule handler is written with a way to track request it has handled so repetition can be avoided 
2. Allow the rule handlers means to call the remote API to interrogate whether the trigger has been dealt with.

**Comment by ArcticSnowman** [comment]
@wass3r  - Thought I'd start a discussion (rather than a ticket) to track discussion on this feature

**Comment by wass3r** [comment]
so for option 1, things get complicated when you have multiple instances of the bot. i know a few deployments that have multiple instances for reliability.

for now, i think option 2 is the most straight-forward. we should have access to all the arguments required to make the call to https://api.slack.com/methods/conversations.replies when we get an event.

some things to consider. we might have to differentiate between events received in threads vs top-level?

i think then we just iterate (if needed, page) through the replies and find our message (might also pose some challenges depending on what the bot is "programmed" to post), or make it more crude and just abort if there was ANY activity from the bot in the thread.
